### PR TITLE
remove liquidity example

### PR DESCRIPTION
**Pull request by chappie1998**
## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

**Comment by chappie1998** [comment]
added an example of removing liquidity.

File: examples/remove_liquidity.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
...ility":"nonpayable","type":"function"},{"constant":true,"inputs":[],"name":"token0","outputs":[{"internalType":"address","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"token1","outputs":[{"internalType":"address","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"totalSupply","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"internalType":"address","name":"to","type":"address"},{"internalType":"uint256","name":"value","type":"uint256"}],"name":"transfer","outputs":[{"internalType":"bool","name":"","type":"bool"}],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"address","name":"from","type":"address"},{"internalType":"address","name":"to","type":"address"},{"internalType":"uint256","name":"value","type":"uint256"}],"name":"transferFrom","outputs":[{"internalType":"bool","name":"","type":"bool"}],"payable":false,"stateMutability":"nonpayable","type":"function"}]"#)?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:18]: given that the example only calls two functions I'd rather prefer a simple abigen! human readable example than pulling in the complete abi here

File: examples/remove_liquidity.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        let receipt = pending_tx.confirmations(6).await?;
+        println!("contract approved succesfully!");
+        println!("{:?}", receipt);
+
+        println!("Removing {} liquidity!", &liquidity);
+        let contract_with_wallet = swap_contract.connect(Arc::clone(&client));
+
+        let tokena = "0xc843a84f06b55664955fc1b2d368d7c4b1dff8f5".parse::<Address>()?;
+        let tokenb = "0x082442ad36236505c63597b5bae89bdf6298a5a9".parse::<Address>()?;
+        let amounta_min = U256::from_dec_str("0")?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:77]:
```suggestion
        let amounta_min = U256::zero();
```

File: examples/remove_liquidity.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        let amountb_min = U256::from_dec_str("0")?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:78]:
```suggestion
        let amountb_min = U256::zero();
```

File: examples/remove_liquidity.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        
+        let contract_with_wallet = pair_contract.connect(Arc::clone(&client));
+        
+        let liquidity = U256::from_dec_str("100")?;
+
+        let call = contract_with_wallet
+            .method::<_, H256>("approve", (swap_address, liquidity.clone()))?;
+
+        let pending_tx = call.send().await?;
+        let receipt = pending_tx.confirmations(6).await?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:68]: 6 confirmations seem excessive in an example

File: examples/remove_liquidity.rs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    println!("token price in % = {}", price);
+
+    // if (price > 10){
+        println!("Apporoving the transaction!");
+        
+        let contract_with_wallet = pair_contract.connect(Arc::clone(&client));
+        
+        let liquidity = U256::from_dec_str("100")?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:62]:
```suggestion
        let liquidity = U256::zero();
```

File: examples/remove_liquidity.rs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    if reserve0 > reserve1 {
+        price = reserve0/reserve1;
+    } else{
+        price = reserve1/reserve0;
+    }
+
+    println!("token price in % = {}", price);
+
+    // if (price > 10){
```
Comment (mattsse) [inline examples/remove_liquidity.rs:57]:
```suggestion
```

File: examples/remove_liquidity.rs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+        println!("Removing {} liquidity!", &liquidity);
+        let contract_with_wallet = swap_contract.connect(Arc::clone(&client));
+
+        let tokena = "0xc843a84f06b55664955fc1b2d368d7c4b1dff8f5".parse::<Address>()?;
+        let tokenb = "0x082442ad36236505c63597b5bae89bdf6298a5a9".parse::<Address>()?;
+        let amounta_min = U256::from_dec_str("0")?;
+        let amountb_min = U256::from_dec_str("0")?;
+        let owner_aadress = "0xF6A996Ce046f5B65C2C3183e9BcbE22d001441F2".parse::<Address>()?;
+        let deadline = U256::from_dec_str("1950886733")?;
```
Comment (mattsse) [inline examples/remove_liquidity.rs:80]:
```suggestion
        let deadline: U256 = 1950886733u64.into();
```

File: examples/remove_liquidity.rs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+   
+   let pair_address = "0xA6108E4d436bE592bAc12F9A0aB7D9A10d821176".parse::<Address>()?;
+   let swap_address = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D".parse::<Address>()?;
+
+//    let liquidity = U256::from_dec_str("50000000")?;
+//    println!("data {}", liquidity);
```
Comment (mattsse) [inline examples/remove_liquidity.rs:27]:
```suggestion
```

File: examples/remove_liquidity.rs [hunk 9]
```diff
@@ -0,0 +1,94 @@
+use ethers::{core::{
```
Comment (mattsse) [inline examples/remove_liquidity.rs:1]: let's add brief description what this example does

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks for this, 
couple nits before we can add this

File: examples/remove_liquidity.rs (hunk 2 above)
Comment (chappie1998) [reply examples/remove_liquidity.rs]: done

File: examples/remove_liquidity.rs (hunk 3 above)
Comment (chappie1998) [reply examples/remove_liquidity.rs]: done

File: examples/remove_liquidity.rs (hunk 4 above)
Comment (chappie1998) [reply examples/remove_liquidity.rs]: done

**Review by chappie1998 — COMMENTED** [review]
Made all the necessary changes except ABI. There are a lot of functions that can be used. For example, adding liquidity and swapping tokens that's why I kept the ABI.

**Review by mattsse — APPROVED** [review]
lgtm

**Review by gakonst — CHANGES_REQUESTED** [review]

**Comment by chappie1998** [comment]
Changes are done. I intentionally added ABI in this form as I wanted to create an ABI system similar to  ether.js. The user can directly use the whole ABI of the contract from explorer without writing the ABI in human-readable form. The user can use any contract. Also, if the user wants to use other functions as well, there is no need to write the ABI in human-readable form every time. If you want, I can put this ABI in a JSON file.

File: examples/remove_liquidity.rs [hunk 10]
```diff
@@ -10,8 +10,8 @@ use eyre::Result;
 
 
 // Remove liquidity from uniswap V2. 
-// This example will remove 500 liquidity of 2 test tokens, TA and TB on rinkbay testnet.
-// THis example uses pair contract and uniswap swap contract to remove liquidity.   
+// This example will remove 500 liquidity of 2 test tokens, TA and TB on Rinkbay testnet.
```
Comment (mattsse) [inline examples/remove_liquidity.rs:13]:
```suggestion
// This example will remove 500 liquidity of 2 test tokens, TA and TB on Rinkeby testnet.
```

**Review by gakonst — APPROVED** [review]
redid the PR to be a little cleaner, thanks for the contribution!

**Pull request merged**